### PR TITLE
Make sure that the missing fields checks is done after the query is built

### LINKED ISSUE #6606: select in contain on deep association
Hi, i'm trying to select a field in deep association and it fail:

```
$item = $this->Items->find('all')
  ->contain(['Elements.Values' => function ($q) {
    return $q->select(['Values.name']);
  })->toArray();
```

The "values" property of Elements entity is an empty array but if i remove the callback function "values"  is well defined by an array of all "Values" entities associated with "Elements"

I tried the code below and it works fine:
```
$item = $this->Items->find('all')
  ->contain(['Elements' => function ($q) {
    return $q->select(['Elements.name']);
  })->toArray();
```

What goes wrong ? Deep associations ? Type of association ? Or both ?

### PR DESCRIPTION
**Pull request by lorenzo**
After this changes, when using the query builder function for a HasMany or
BelongsToMany association, failing to select the foreignKey fields will
actually trigger the helpful exception for the user.

Fixes #6606

### CODE REVIEW

**Comment by ADmad** [comment]
Good idea. Tests failing.

File: src/ORM/Association/SelectableAssociationTrait.php [hunk 1]
```diff
@@ -82,8 +82,11 @@ protected function _buildQuery($options)
 
         $finder = isset($options['finder']) ? $options['finder'] : $this->finder();
         list($finder, $opts) = $this->_extractFinder($finder);
+        $options += ['fields' => []];
+
         $fetchQuery = $this
             ->find($finder, $opts)
+            ->select($options['fields'])
```
Comment (markstory) [inline src/ORM/Association/SelectableAssociationTrait.php]: How will this interact with autoFields()?

Comment (lorenzo) [inline src/ORM/Association/SelectableAssociationTrait.php]: No changes to previous behavior. That call was done before later in that method. I just added it here for clarity

File: src/ORM/Association/SelectableAssociationTrait.php [hunk 2]
```diff
@@ -117,10 +110,43 @@ protected function _buildQuery($options)
...
+     */
+    protected function _assertFieldsPresent($fetchQuery, $key)
+    {
+        $select = $fetchQuery->aliasFields($fetchQuery->clause('select'));
+        if (empty($select)) {
+            return;
+        }
+        $missingFields = array_diff($key, $select) !== [];
+
+        if ($missingFields) {
```
Comment (jippi) [inline src/ORM/Association/SelectableAssociationTrait.php]: change to !$missingFields { return } to avoid nesting of code :)

Comment (lorenzo) [inline src/ORM/Association/SelectableAssociationTrait.php]: In this case I cannot do this

**Comment by lorenzo** [comment]
Tests are passing now

**Pull request merged**